x-pack/plugin/core/src/main/resources/new-security-index-template.json

```diff
@@ -0,0 +1,151 @@
+{
```

Comment:
why is this a new file?

Reply:
This will be present in 6.7 only. When I backport to 7 `new-security-index-template.json` will take the place of `security-index-template.json` .

The rationale is as follows:
In 6.7 nodes will create `.security-6` as usual, if the index is missing (fresh install).
`_xpack/migration/upgrade/.security-6` should be called before rolling upgrading to 7.
The alternative could have been to create `.security-7` on a fresh 6.7 install, but during a rolling upgrade from 6.6 to 6.7, 6.7 nodes might erroneously consider they have to create `.security-7`.

Reply:
> This will be present in 6.7 only. When I backport to 7

This phrasing confuses me a little - what I assume you mean is that once you've backported this change to 6.7 you will update master (7.0) to have only a single template (rename `new-security-index-template.json` to `security-index-template.json`).

I think in that case it might be better to name the json files as  `security-index-template-6.json` and `security-index-template-7.json`